src/widgets/SensorStatus.js

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect } from 'react';
+import { Animated, TouchableOpacity, Text } from 'react-native';
+import PropTypes from 'prop-types';
+
+import { FlexRow } from './FlexRow';
+import { FlexView } from './FlexView';
+import { LowBatteryIcon, HotBreachIcon, ColdBreachIcon } from './icons';
+
+import { DARKER_GREY, APP_FONT_FAMILY, COLD_BREACH_BLUE, WHITE, DANGER_RED } from '../globalStyles';
+import { MILLISECONDS } from '../utilities';
+import { Rectangle } from './Rectangle';
+import temperature from '../utilities/temperature';
+
+const BigText = ({ children, colour }) => (
+  <Text style={{ fontSize: 32, fontFamily: APP_FONT_FAMILY, color: colour }}>{children}</Text>
+);
+
+BigText.defaultProps = { colour: WHITE };
+BigText.propTypes = { children: PropTypes.node.isRequired, colour: PropTypes.string };
+
+export const SensorStatus = ({ isInHotBreach, isInColdBreach, isLowBattery, currentTemp }) => {
+  const fadeAnim1 = React.useRef(new Animated.Value(0)).current;
+  const fadeAnim2 = React.useRef(new Animated.Value(0)).current;
+  const fadeAnim3 = React.useRef(new Animated.Value(0)).current;
+  const animRef = React.useRef(null);
+
+  const conditions = [isInHotBreach, isInColdBreach, isLowBattery, !!currentTemp != null];
+  const animationValues = [fadeAnim1, fadeAnim2, fadeAnim3].filter((_, i) => conditions[i]);
+  const isInDanger = isInHotBreach || isInColdBreach || isLowBattery;
+
+  const stopAnimating = () => animRef?.current?.stop();
+
+  const startAnimating = () => {
+    // If there is a ref to an animation, stop it before starting
+    // a new one.
+    stopAnimating();
+
+    // Start an infinite looping sequence of animations.
+    animRef.current = Animated.loop(
+      // Sequence is a a sequence of sequences - each sub sequence being
+      // for a specific icon which should be shown. i.e. if the fridge is
+      // low battery and has a hot breach, there is a sequence of sequences
+      // for each icon.
+      Animated.sequence(
+        // Map each animation value (which can be thought of as an icon)
+        // to a sequence which will increase the opacity of the icon to full
+        // over one second, then reduce the opacity to zero over one second.
```

Comment:
I don't really like so much commenting I think it makes the code harder to read. Please let me know if it is too much as a reader or if it was helpful.